File: q1/q1.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int first = 0,last = s.size();
        vector<int> arr(s.size()+1,0);
        for(int i=0;i<=s.size();++i)
        {   if(s[i]=='I')
                arr[i]=first++;
            else
                arr[i]=last--;
        }
        arr[s.size()]=first;
        return arr;
    }
};

File: q1/q1.py


class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        first = 0
        last = len(s)
        new = []
        for i in s:
            if i=='D':
                new.append(last)
                last-=1
            else:
                new.append(first)
                first+=1
                
        new.append(first)
        return new